Userland/Libraries/LibWeb/Painting/BorderPainting.cpp

```diff
@@ -277,101 +564,40 @@ void paint_all_borders(PaintContext& context, CSSPixelRect const& bordered_rect,
-            return corner_color.with_alpha((corner_color.alpha() * corner_pixel.alpha()) / 255);
-        });
-    };
-
-    // FIXME: Corners should actually split between the two colors, if both are provided (and differ)
-    auto pick_corner_color = [](auto const& border, auto const& adjacent_border) {
-        if (border.width > 0)
-            return border.color;
-        return adjacent_border.color;
-    };
-
-    // Blit the corners into to their corresponding locations:
-    if (top_left)
-        blit_corner(border_rect.top_left().to_type<int>(), top_left.as_rect(), pick_corner_color(borders_data.top, borders_data.left));
-
-    if (top_right)
-        blit_corner(border_rect.top_right().to_type<int>().translated(-top_right.horizontal_radius, 0), top_right.as_rect().translated(corner_mask_rect.width().value() - top_right.horizontal_radius, 0), pick_corner_color(borders_data.top, borders_data.right));
+    AK::CircularQueue<BorderEdge, 4> borders;
+    borders.enqueue(BorderEdge::Top);
+    borders.enqueue(BorderEdge::Right);
+    borders.enqueue(BorderEdge::Bottom);
+    borders.enqueue(BorderEdge::Left);
+
+    // try to find the first border that has a different color with previous one.
+    // then start painting from that border.
+    for (size_t i = 0; i < borders.capacity(); i++) {
+        if (border_color(borders.at(0), borders_data) != border_color(borders.at(1), borders_data)) {
+            borders.enqueue(borders.dequeue());
+            break;
+        }
 
-    if (bottom_right)
-        blit_corner(border_rect.bottom_right().to_type<int>().translated(-bottom_right.horizontal_radius, -bottom_right.vertical_radius), bottom_right.as_rect().translated(corner_mask_rect.width().value() - bottom_right.horizontal_radius, corner_mask_rect.height().value() - bottom_right.vertical_radius), pick_corner_color(borders_data.bottom, borders_data.right));
+        borders.enqueue(borders.dequeue());
+    }
 
-    if (bottom_left)
-        blit_corner(border_rect.bottom_left().to_type<int>().translated(0, -bottom_left.vertical_radius), bottom_left.as_rect().translated(0, corner_mask_rect.height().value() - bottom_left.vertical_radius), pick_corner_color(borders_data.bottom, borders_data.left));
+    Gfx::Path path;
+    for (BorderEdge edge : borders) {
+        switch (edge) {
+        case BorderEdge::Top:
+            paint_border(context, BorderEdge::Top, top_border_rect, top_left, top_right, borders_data, path, edge == borders.last());
+            break;
+        case BorderEdge::Right:
+            paint_border(context, BorderEdge::Right, right_border_rect, top_right, bottom_right, borders_data, path, edge == borders.last());
+            break;
+        case BorderEdge::Bottom:
+            paint_border(context, BorderEdge::Bottom, bottom_border_rect, bottom_right, bottom_left, borders_data, path, edge == borders.last());
+            break;
+        default: // BorderEdge::Left:
```

Comment:
```suggestion
        case: BorderEdge::Left
```
(make default: `VERIFY_NOT_REACHED()`)